internal/api/mfa.go

```diff
@@ -217,10 +503,25 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 	if err := retrieveRequestParams(r, params); err != nil {
 		return err
 	}
+
+	switch factor.FactorType {
+	case models.Phone:
+		if !config.MFA.Phone.VerifyEnabled {
+			return unprocessableEntityError(ErrorCodeMFAPhoneEnrollDisabled, "MFA verification is disabled for SMS")
+		}
+		return a.verifyPhoneFactor(w, r, params)
+	case models.TOTP:
+		if !config.MFA.TOTP.VerifyEnabled {
+			return unprocessableEntityError(ErrorCodeMFATOTPEnrollDisabled, "MFA verification is not enabled for TOTP")
+		}
+	default:
+		return badRequestError(ErrorCodeValidationFailed, "factor_type needs to be totp or sms")
+	}
+
 	currentIP := utilities.GetIPAddress(r)
 
 	if !factor.IsOwnedBy(user) {
-		return internalServerError(InvalidFactorOwnerErrorMessage)
+		return forbiddenError(InvalidFactorOwnerErrorMessage, "MFA Factor not owned by user")
```

Comment:
IMO this should be a 404 Not Found with 0 information leaked whether it's owned by user or not.